src/features/broker-protection/extractors/address.js

```diff
@@ -0,0 +1,73 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { Extractor } from '../types.js'
+import { stringToList } from '../actions/extract.js'
+import parseAddress from 'parse-address'
+import { states } from '../comparisons/constants.js'
+
+/**
+ * @implements {Extractor<{city:string; state: string|null}[]>}
+ */
+export class CityStateExtractor {
+    /**
+     * @param {string[]} strs
+     * @param {import('../actions/extract.js').ExtractorParams} extractorParams
+     */
+    extract (strs, extractorParams) {
+        const cityStateList = strs.map(str => stringToList(str, extractorParams.separator)).flat()
+        return getCityStateCombos(cityStateList)
+    }
+}
+
+/**
+ * @implements {Extractor<{city:string; state: string|null}[]>}
+ */
+export class AddressFullExtractor {
```

Comment:
This was just moved from the original place so that it can live alongside the extractor